app/javascript/app/components/modal/modal-component.jsx

```diff
@@ -34,19 +35,21 @@ class CustomModal extends PureComponent {
     const modalStyles = { ...defaultStyles, ...customStyles };
 
     return (
-      <Modal
-        className={{ base: styles.modal }}
-        style={modalStyles}
-        isOpen={isOpen}
-        contentLabel={contentLabel}
-        shouldCloseOnOverlayClick={shouldCloseOnOverlayClick}
-      >
-        {header}
-        <Button onClick={onRequestClose} className={theme.closeBtn} square>
-          <Icon icon={closeIcon} className={theme.close} />
-        </Button>
-        <div className={theme.content}>{children}</div>
-      </Modal>
+      <ClickOutside onClickOutside={onRequestClose}>
```

Comment:
We don't need another dependency to attach a listener to click outside because the library already has one, please review this example http://reactcommunity.org/react-modal/examples/should_close_on_overlay_click.html#using-shouldcloseonoverlayclick and update our own version with that feature

Reply:
👍 Thanks!